server.js

```diff
@@ -1,33 +1,117 @@
+import cors from 'cors';
+
+// My choosen dataset
+import books from './data/books.json';
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
-//
+
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello, welcome to Gabriella´s API with books! Possible endpoints to use: /books (for seeing all books), /books/book/:id, (to search by Book ID) /books/author/:author (to search by author) and /books/top-rated (to see the 20 highest rated books) With filter and query parameter you can find search =high to get 50 books with high rating and you can use page=pagenumber to se 50 per page where it exist 10 pages in total')
+});
+
+// The first endpoint, returns a collection of books
+app.get('/books', (req, res) => {
+  res.json(books)
+});
+
+// Filter query
+app.get('/books/filter', (req, res) => {
+  // Slice to set the 50 first books
+  let filteredBooks = books.slice(0, 50);
+
+  // Set a filter with query parameter to filter on
+  // only the books with average_rating > 4.0
+  // Access this query by '/books?average_rating=high'
+  const { average_rating, page } = req.query;
+  if (average_rating === 'high') {
+    filteredBooks = filteredBooks.filter((item) => item.average_rating > 4.0);
+  };
+
+  // Set query for which site to see, with 50 on each.
+  // Total 500 books divided on 10 pages
+  if (page === '1') {
+    filteredBooks = books.slice(0, 50)
+  } else if (page === '2') {
+    filteredBooks = books.slice(51, 100)
+  } else if (page === '3') {
+    filteredBooks = books.slice(101, 150)
+  } else if (page === '4') {
+    filteredBooks = books.slice(151, 200)
+  } else if (page === '5') {
+    filteredBooks = books.slice(201, 250)
+  } else if (page === '6') {
+    filteredBooks = books.slice(251, 300)
+  } else if (page === '7') {
+    filteredBooks = books.slice(301, 350)
+  } else if (page === '8') {
+    filteredBooks = books.slice(351, 400)
+  } else if (page === '9') {
+    filteredBooks = books.slice(401, 450)
+  } else if (page === '10') {
+    filteredBooks = books.slice(451, 500)
+  }
+
+  res.json(filteredBooks)
+});
+
+// The second endpoint, returns a single book based on bookID
+app.get('/books/:id', (req, res) => {
+  const bookID = req.params.id;
+  const bookById = books.find((item) => item.bookID === +bookID);
+
+  // If bookID wasn't found
+  if (!bookById) {
+    res.send("Sorry, we cannot find any book with that ID :( Please try another ID!");
+  };
+
+  res.json(bookById);
+});
+
+// Returns all books per author /books/authorname
+app.get('/books/author/:author', (req, res) => {
```

Comment:
Could we make this more user friendly by allowing only part of the name and case insensitivity?

Reply:
Also, maybe this could be a query under the `/books` endpoint as well, since it is returning books and not authors :)